projects/step-frontend/src/lib/modules/timeseries/modules/chart/injectables/tooltip-plugin.ts

```diff
@@ -16,62 +16,13 @@ export class TooltipPlugin {
   private win = this._doc.defaultView!;
   private _executionsService = inject(ExecutionsService);
 
-  private createTooltipElement() {
-    const tooltip = this.createElementWithClass('div', 'ts-tooltip');
-    tooltip.id = 'tooltip';
-    tooltip.classList.add('ts-tooltip');
-    tooltip.style.display = 'none';
-    tooltip.style.position = 'absolute';
-    return tooltip;
-  }
-
-  private createExecutionsMenu = (event: MouseEvent, tooltipContainer: HTMLElement, executionIds: string[]) => {
-    const menu = this.createElementWithClass('div', 'tooltip-menu');
-    menu.innerText = 'Loading...';
-    const tooltipBounds = tooltipContainer.getBoundingClientRect();
-    let computedLeft = event.clientX - tooltipBounds.left + 16;
-    const computedTop = event.clientY - tooltipBounds.top + 4;
-    const estimatedWidthOfExecutionLabel = 150;
-    if (event.clientX + 16 + estimatedWidthOfExecutionLabel > this.win.innerWidth) {
-      computedLeft -= estimatedWidthOfExecutionLabel;
-    }
-    menu.style.left = computedLeft + 'px';
-    menu.style.top = computedTop + 'px';
-    this._executionsService.getExecutionsByIds(executionIds).subscribe((executions) => {
-      menu.innerText = '';
-      executions.forEach((ex) => {
-        const row = this.createElementWithClass('div', 'link-row');
-        row.setAttribute('title', 'Show executions');
-        row.innerText = `${ex.description!} (${new Date(ex.startTime!).toLocaleString()})`;
-        row.addEventListener('click', () => {
-          this.win.open(getExecutionLink(ex.id!));
-        });
-        menu.appendChild(row);
-      });
-      let menuRightBound = menu.getBoundingClientRect().right;
-      if (menuRightBound > this.win.innerWidth) {
-        menu.style.left = computedLeft - (menuRightBound - this.win.innerWidth) - 16 + 'px';
-      }
-    });
-
-    return menu;
-  };
-
-  private createRowLeftSection(row: TooltipRowEntry) {
-    const leftContainer = this.createElementWithClass('div', 'left');
-    const nameDiv = this.createElementWithClass('div', 'name');
-    nameDiv.textContent = `${row.name} `;
-    if (row.color) {
-      const colorDiv = this.createElementWithClass('div', 'color');
-      colorDiv.style.backgroundColor = row.color;
-      leftContainer.appendChild(colorDiv);
-    }
-    leftContainer.appendChild(nameDiv);
-    return leftContainer;
-  }
-
+  /**
+   * The tooltip will be displayed while the user will hover the chart.
+   * An execution is optional to display. It can be configured via settings and via chart metadata (where the actual
+   * data is taken from)
+   */
```

Comment:
This sounds more like infor for the KB than a needed comment in the code?

Reply:
@neogucky it's for us. In order to have the execution link working there are a few things to configure - we had a bug for it and I had to investigate pretty much to remember why it is causing. It's important to mention that "settings and metadata" are needed for this.

Reply:
I can simplify it though